ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/lib.test.ts

```diff
@@ -122,4 +156,145 @@ describe("lib", () => {
       expect(result.length).toBe(5);
     });
   });
+
+  describe("deleteSuccessfullyParsedMessages", () => {
+    it('calls queueClient.deleteMessage appropriately', async () => {
+      // GIVEN
+      const queueClientMock: QueueClient = {
+        deleteMessage: jest.fn().mockResolvedValue(true),
+      } as any; 
+      const messages: DequeuedMessageItem[] = [{
+        messageId: '1234',
+        popReceipt: 'abcd'
+      },{
+        messageId: '1234',
+        popReceipt: 'abcd'
+      },{
+        messageId: '1234',
+        popReceipt: 'abcd'
+      }] as any;
+
+      // WHEN
+      await deleteSuccessfullyParsedMessages(context, queueClientMock, messages, {});
+      
+      // THEN
+      expect(queueClientMock.deleteMessage).toHaveBeenCalledTimes(messages.length);
+    });
+
+    it("does't call queueClient.deleteMessage for parse failures", async () => {
+      // GIVEN
+      const queueClientMock: QueueClient = {
+        deleteMessage: jest.fn().mockResolvedValue(true),
+      } as any; 
+      const messages: DequeuedMessageItem[] = [{
+        messageId: 'apple',
+        popReceipt: 'abcd'
+      },{
+        messageId: 'banana',
+        popReceipt: 'abcd'
+      },{
+        messageId: 'grape',
+        popReceipt: 'abcd'
+      }] as any;
+      const parseFailure = {
+        'grape': true
+      };
+
+      // WHEN
+      await deleteSuccessfullyParsedMessages(context, queueClientMock, messages, parseFailure);
+      
+      // THEN
+      expect(queueClientMock.deleteMessage).toHaveBeenCalledTimes(messages.length - 1);
```

Comment:
we can be more specific here and add a to verify that the failed one is not being deleted 
` expect(queueClientMock.deleteMessage).not.toHaveBeenCalledWith( grape );`

Reply:
Added in cc49151